### PR TITLE
Set CA for adding and deleting contacts, as well as changing the DAO

### PR DESCRIPTION
**Pull request by Ahassini04**
For the DAO I made it such that it now holdsand attribute of the logged in user entity which can be used in the different use cases.

### CODE REVIEW

File: src/main/java/interface_adapter/mutating_contacts/MutatingContactsController.java
Comment (az19642) [inline src/main/java/interface_adapter/mutating_contacts/MutatingContactsController.java:1]: Good. This class acts as the controller for the specific use case and depends on an Input Boundary and creates Input Data to execute the Input Boundary (interactor) with.

File: src/main/java/use_case/mutating_contacts/MutatingContactsInputBoundary.java
Comment (az19642) [inline src/main/java/use_case/mutating_contacts/MutatingContactsInputBoundary.java:1]: Good. Input Boundary contains only the execute method, adhering to ISP.

File: src/main/java/use_case/mutating_contacts/MutatingContactsInputData.java
Comment (az19642) [inline src/main/java/use_case/mutating_contacts/MutatingContactsInputData.java:1]: Good. Input Data contains all the necessary data to be used in the interactor.

File: src/main/java/use_case/mutating_contacts/MutatingContactsInteractor.java
Comment (az19642) [inline src/main/java/use_case/mutating_contacts/MutatingContactsInteractor.java:1]: Good. Manipulates data using the Data Access Interface (implemented by MongoUserDataAccessObject) and calls the Presenter appropriately with the required output data (in this case, none for the success view and error messages for the fail view)

File: src/main/java/use_case/mutating_contacts/MutatingContactsUserDataAccessInterface.java
Comment (az19642) [inline src/main/java/use_case/mutating_contacts/MutatingContactsUserDataAccessInterface.java:1]: Good. Follows ISP. All the specific methods required by the use case is in this interface while the actual data access object implements this (of many).

**Review by az19642 — CHANGES_REQUESTED** [review]
Good implementation overall of the the adding/removing contacts use case. Interactor test is flawed.

**Coding style** 
Good. Private and public access modifiers are used where appropriate and the files created for the use case follow the clean architecture packaging.

**Correctness**
Correctness cannot be tested by running the application (non-testing) as the listeners for this use case have not been implemented yet.

**Tests**
The interactor test is invalid and produces an error. You must rethink how you will test this interactor since your MutatingContactsUserDataAccessInterface does not specify a method to set the current user so the current user is null during your test.

**Clean Architecture + SOLID**
Good. All files mimic clean architecture, inherently adhering to SOLID.

**Javadoc**
Good. All public classes and methods were properly documented and explains the point of the class/method.

**Clarity**
PR description can be more descriptive. Explain the reason for the DAO being changed this way. Could also elaborate on the actual use case itself, e.g.: why did you decide to do adding/removing contacts in one use case?

File: src/test/java/use_case/mutating_contacts/MutatingContactsInteractorTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class MutatingContactsInteractorTest {
+    @Test
+    void execute() {
+        MutatingContactsInputData inputData = new MutatingContactsInputData("test", true);
+        MutatingContactsUserDataAccessInterface mutatingContactsUserDataAccessInterface = new MongoUserDataAccessObject(
+                System.getenv("MONGO_PASSWORD"),
+                new CommonUserFactory()
```
[the remark this answers is not in the document]
Comment (jchrisoh) [reply src/test/java/use_case/mutating_contacts/MutatingContactsInteractorTest.java:15]: Resolved.

**Comment by az19642** [comment]
Looks good, interactor test is correct and passes.

**Review by az19642 — APPROVED** [review]

**Pull request merged**